network/tcp_test.go

```diff
@@ -363,10 +363,48 @@ func TestTCPConnWithListener(t *testing.T) {
 	}
 }
 
-// will create a TCPListener & open a golang net.TCPConn to it
+// will create a TCPListener with a specific address to listen on &
+// open a golang net.TCPConn to it
+func TestTCPListenerWithListenAddr(t *testing.T) {
+	addr := NewAddress(PlainTCP, "127.0.0.1:5678")
```

Comment:
A test that listens on a fixed port will give false negatives if that port is already in use. Fragile tesdt = crazy mad Jeff. :)

Instead, listen on port 0, so that the kernel gets to choose. Use "addr := NewAddress(PlainTCP, "1.2.3.4:0") and listenAddr := net.ParseIP("127.0.0.1")". Then connect to it via 127.0.0.1 (use svr.Address() to find out the address that the kernel assigned). If you succeed, it means the listenAddr worked (otherwise, it would be trying and failing to listen on 1.2.3.4).

Reply:
Alright, makes sense. In this case, should we also change the other tests in this file listening on a fixed port (e.g. TestTCPListener, TestTCPConnWithListener, TestTCPConn)? If yes, should I do it for this PR?